### PR TITLE
Add translations for unit types on listing directory page DAH-1053

### PR DESCRIPTION
**Pull request by jacksongonzales**
DAH-1053

Add translations for: 

* Unit types (1BR, Studio)

* 1 BR should be updated to be “1 bedroom”

* Studio can be translated

* Update SRO to "SRO (single-room occupancy)"

### CODE REVIEW

File: app/assets/json/translations/react/en.json [hunk 1]
```diff
@@ -64,10 +64,12 @@
   "listings.reservedCommunityType.habitat": "Habitat Greater San Francisco",
   "listings.reservedCommunityType.senior": "Senior Building",
   "listings.reservedCommunityType.veteran": "Veterans Building",
-  "listings.unitTypes.1 BR": "1 BR",
-  "listings.unitTypes.2 BR": "2 BR",
-  "listings.unitTypes.3 BR": "3 BR",
-  "listings.unitTypes.SRO": "SRO",
+  "listings.unitTypes.1 BR": "1 Bedroom",
```
Comment (akegan) [inline app/assets/json/translations/react/en.json:67]: I think we don't want these spaces in the keys, I'm not sure that it'll work with Phrase with the type of files we have set up.

It is a nice solution, so we could try it to see if it'll work OK.

File: app/assets/json/translations/react/en.json [hunk 2]
```diff
@@ -64,10 +64,12 @@
...
+  "listings.unitTypes.2 BR": "2 Bedroom",
+  "listings.unitTypes.3 BR": "3 Bedroom",
+  "listings.unitTypes.4 BR": "4 Bedroom",
+  "listings.unitTypes.5 BR": "5 Bedroom",
+  "listings.unitTypes.SRO": "SRO (single-room occupancy)",
```
Comment (akegan) [inline app/assets/json/translations/react/en.json:72]: SRO should just be SRO for now. We don't have room for the explanation in the parentheses in the tables we have

File: app/javascript/__tests__/util/languageUtil.test.ts [hunk 3]
```diff
@@ -160,4 +161,13 @@ describe("languageUtil", () => {
       expect(getReservedCommunityType("New Type")).toBe("New Type")
     })
   })
+  describe("emptyIfNotTranslated", () => {
+    it("returns a translated string", () => {
+      expect(emptyIfNotTranslated("listings.unitTypes.2 BR")).toBe("2 Bedroom")
+    })
+
+    it("returns an empty string when no translation", () => {
+      expect(emptyIfNotTranslated("listings.unitTypes.500 BR")).toBe("")
```
Comment (akegan) [inline app/javascript/__tests__/util/languageUtil.test.ts:170]: We want the behavior to be to show whatever salesforce is sending if we don't have a translation for the value. So we don't require a code change if a new type of unit is added

**Review by akegan — CHANGES_REQUESTED** [review]
Couple of minor changes.

File: app/assets/json/translations/react/en.json (hunk 1 above)
Comment (jacksongonzales) [reply app/assets/json/translations/react/en.json:67]: right, @christine-sfg and I agreed that it's funky. if this works as a first pass we could dogear it as something to come back to

File: app/javascript/pages/RentDirectory.tsx [hunk 4]
```diff
@@ -38,8 +38,7 @@ const getForRentSummaryTable = (listing: RailsRentalListing) =>
     .filter((summary) => !!summary.unitType)
     .map((summary) => ({
       unitType: {
-        cellText:
-          emptyIfNotTranslated(`listings.unitTypes.${summary.unitType}`) || summary.unitType,
+        cellText: emptyIfNotTranslated(`listings.unitTypes.${summary.unitType}`, summary.unitType),
```
Comment (akegan) [inline app/javascript/pages/RentDirectory.tsx:41]: this looks like a typo maybe, needs to be renamed to defaultIfNotTranslated?

Comment (jacksongonzales) [reply app/javascript/pages/RentDirectory.tsx:41]: oop, thanks for catching. still getting the hang of vscode and things like...saving files 😬

**Comment by jacksongonzales** [comment]
@akegan would you mind sending me an invite to CircleCI? I'm guessing I can't rerun the failing tests with no account

**Review by akegan — APPROVED** [review]
this looks good to me. Feel free to squash merge once tests pass (I just kicked them off again)

**Pull request merged**